crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -398,6 +476,78 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
     ///
     /// There must be two descriptors in the chain, both writable: a header descriptor and a data
     /// descriptor.
+    ///
+    /// # Arguments
+    /// * `mem` - the `GuestMemory` object that can be used to access the queue buffers.
+    /// * `desc_chain` - the descriptor chain corresponding to a packet.
+    /// * `max_data_size` - the maximum size allowed for the packet payload, that was negotiated
+    ///                     between the device and the driver. Tracking issue for defining this
+    ///                     feature in virtio-spec
+    ///                     [here](https://github.com/oasis-tcs/virtio-spec/issues/140).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use virtio_queue::defs::VIRTQ_DESC_F_WRITE;
+    /// # use virtio_queue::mock::MockSplitQueue;
+    /// # use virtio_queue::{Descriptor, Queue, QueueState, QueueStateT};
+    /// use virtio_vsock::packet::{VsockPacket, PKT_HEADER_SIZE};
+    /// # use vm_memory::{Bytes, GuestAddress, GuestAddressSpace, GuestMemoryMmap};
+    ///
+    /// # const MAX_PKT_BUF_SIZE: u32 = 64 * 1024;
+    /// # const SRC_CID: u64 = 1;
+    /// # const DST_CID: u64 = 2;
+    /// # const SRC_PORT: u32 = 3;
+    /// # const DST_PORT: u32 = 4;
+    /// # const LEN: u32 = 16;
+    /// # const TYPE_STREAM: u16 = 1;
+    /// # const OP_RW: u16 = 5;
+    /// # const FLAGS: u32 = 7;
+    /// # const FLAG: u32 = 8;
+    /// # const BUF_ALLOC: u32 = 256;
+    /// # const FWD_CNT: u32 = 9;
+    ///
+    /// let mem = GuestMemoryMmap::<()>::from_ranges(&[(GuestAddress(0), 0x10000)]).unwrap();
+    /// let mut queue = MockSplitQueue::new(&mem, 16);
+    /// let mut q = queue.create_queue(&mem);
```

Comment:
Should we also hide these lines? They are creating the queue using the mock framework. Maybe this can cause some confusion about how it's supposed to be used.

Reply:
I didn't remove them so that it's clear what is `q`, but they are indeed confusing. I changed to using the same approach from the `virtio-queue` examples, and have a separate function for the queue creation. I'm a bit disappointed I forgot about it until now :(.